com.unity.render-pipelines.universal/Runtime/RenderingUtils.cs

```diff
@@ -154,27 +160,36 @@ internal class StereoConstants
         /// <param name="cmd">CommandBuffer to submit data to GPU.</param>
         /// <param name="viewMatrix">View matrix to be set. Array size is 2.</param>
         /// <param name="projectionMatrix">Projection matrix to be set.Array size is 2.</param>
+        /// <param name="cameraProjectionMatrix">Camera projection matrix to be set.Array size is 2. Does not include platform specific transformations such as depth-reverse, depth range in post-projective space and y-flip. </param>
         /// <param name="setInverseMatrices">Set this to true if you also need to set inverse camera matrices.</param>
         /// <returns>Void</c></returns>
-        internal static void SetStereoViewAndProjectionMatrices(CommandBuffer cmd, Matrix4x4[] viewMatrix, Matrix4x4[] projMatrix, bool setInverseMatrices)
+        internal static void SetStereoViewAndProjectionMatrices(CommandBuffer cmd, Matrix4x4[] viewMatrix, Matrix4x4[] projMatrix, Matrix4x4[] cameraProjMatrix, bool setInverseMatrices)
         {
             for (int i = 0; i < 2; i++)
             {
                 stereoConstants.viewProjMatrix[i] = projMatrix[i] * viewMatrix[i];
                 stereoConstants.invViewMatrix[i] = Matrix4x4.Inverse(viewMatrix[i]);
                 stereoConstants.invProjMatrix[i] = Matrix4x4.Inverse(projMatrix[i]);
                 stereoConstants.invViewProjMatrix[i] = Matrix4x4.Inverse(stereoConstants.viewProjMatrix[i]);
+                stereoConstants.invCameraProjMatrix[i] = Matrix4x4.Inverse(cameraProjMatrix[i]);
                 stereoConstants.worldSpaceCameraPos[i] = stereoConstants.invViewMatrix[i].GetColumn(3);
             }
 
             cmd.SetGlobalMatrixArray(UNITY_STEREO_MATRIX_V, viewMatrix);
             cmd.SetGlobalMatrixArray(UNITY_STEREO_MATRIX_P, projMatrix);
             cmd.SetGlobalMatrixArray(UNITY_STEREO_MATRIX_VP, stereoConstants.viewProjMatrix);
+
+            cmd.SetGlobalMatrixArray(UNITY_STEREO_CAMERA_PROJECTION, cameraProjMatrix);
+            cmd.SetGlobalMatrixArray(UNITY_STEREO_WORLD_TO_CAMERA, viewMatrix);
```

Comment:
Looks like UNITY_STEREO_WORLD_TO_CAMERA and UNITY_STEREO_MATRIX_V are the same? Should we just define unity_WorldToCamera to unity_StereoMatrixV in shader instead? By doing that, we could save this extra SetGlobalMatrixArray call and also reduce UnityStereoViewBuffer size to save some bandwidth.

Reply:
Yes, they are redundant in the context of XR pass. I wanted to keep the code symmetrical with the obsolete shadergraph stereo matrices. These matrices would only be meaningful if we had an XR pass that was not the main camera and required to access the main camera view/viewInverse matrices. At the moment, there isn't such a case, as well this C# code above does not differentiate the pass transforms and the main camera transforms.

Reply:
Removed.